NitroxClient/Unity/Helper/DebugUtils.cs

```diff
@@ -7,6 +7,46 @@ namespace NitroxClient.Unity.Helper
 {
     public class DebugUtils
     {
+        private static string prefix = "";
+
+        public static void DumpHierarchy(GameObject obj)
+        {
+            Log.Debug("Analyze gameobject: ");
+            Log.Debug("------------------------------------");
+            StartHierachy(obj);
+            Log.Debug("------------------------------------");
+        }
+
+        private static void StartHierachy(GameObject obj)
+        {
+            prefix = prefix + "    ";
+            Log.Debug(GetInfoOfGamobject(obj));
+            for (int i = 0; i < obj.transform.childCount; i++)
+            {
+                StartHierachy(obj.transform.GetChild(i).gameObject);
+                prefix = prefix.Substring(4);
+            }
+        }
+
+        private static string GetInfoOfGamobject(GameObject obj)
+        {
+            return prefix + obj.name + "   --<>--   " + GetComponentsOfGamobject(obj);
+        }
+
+        private static string GetComponentsOfGamobject(GameObject obj)
+        {
+            string list = "";
+            foreach (Component item in obj.GetComponents<Component>())
+            {
+                list += item.GetType() + " | ";
+            }
+            if (list.Length >= 3)
+            {
+                list = list.Substring(0, list.Length-3);
```

Comment:
Yuck. Use something like `string.Join` to add the separator.

This entire function can be converted into a oneliner:
```C#
return string.Join(" | ", obj.GetComponents<Component>().Select(item => item.GetType().ToString()).ToArray());
```